src/components/Menu/MenuItem.tsx

```diff
@@ -121,51 +124,71 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
      */
     renderIcon: PropTypes.func,
 
+    /**
+     * A custom render function the wrapper slot.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderWrapper: PropTypes.func,
+
     /** Additional CSS styles to apply to the component instance.  */
     styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
 
     /** Override for theme site variables to allow modifications of component styling via themes. */
     variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Shorthand for the wrapper component */
+    wrapper: PropTypes.oneOfType([PropTypes.node, PropTypes.object]),
   }
 
   static defaultProps = {
-    as: 'li',
+    as: 'a',
```

Comment:
when we discussed changes for `Input` component we had a consensus that **`as` and `styles` props always go to the root element**; we might want the same for `MenuItem`; let's see what others say @levithomason @kuzhelov

Reply:
IIRC there is an accessibility requirement that the click listener needs to be on the anchor tag.  The primary use case for `as` on the menu item is to allow adding router functionality.  This means the router's navigation handler needs to be applied to the anchor:

```jsx
import { NavLink } from 'react-router'

<Menu items={[{ as: NavLink, to: '/', content: 'Home' }]} />
```

If the `as` prop goes to the root, it will still work as the event will bubble from the anchor up to the `li`, however, it would not be the correct markup and behavior for the accessibility requirement.